libs/sdk-core/src/error.rs

```diff
@@ -1,60 +1,311 @@
+use std::fmt;
+
+use anyhow::Error;
+use bitcoin::secp256k1;
+use serde::Serialize;
 use thiserror::Error;
 
 pub type SdkResult<T, E = SdkError> = Result<T, E>;
 
+/// The service error codes
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Serialize)]
+pub enum ServiceErrorCode {
+    BackupFailed,
+    ClientError,
+    DerivationFailed,
+    Generic,
+    InitFailed,
+    NotReady,
+    PersistenceFailed,
+    RequestFailed,
+    ShutdownFailed,
+    SignatureError,
+    StreamAlreadyCreated,
+    TransportError,
+}
+
+impl fmt::Display for ServiceErrorCode {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+/// The LSP error codes
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Serialize)]
+pub enum LspErrorCode {
+    CalculateOpenChannelFeesFailed,
+    ChannelOpeningNotAvailable,
+    ConnectionFailed,
+    LspNotFound,
+}
+
+impl fmt::Display for LspErrorCode {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+/// The payment error codes
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Serialize)]
+pub enum PaymentErrorCode {
+    AlreadyPaid,
+    ChannelNotFound,
+    InvalidAmount,
+    OpeningFeeParamsNotFound,
+    PaymentFailed,
+    PaymentNotFound,
+    PaymentTimeout,
+}
+
+impl fmt::Display for PaymentErrorCode {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+/// The onchain error codes
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Serialize)]
+pub enum OnchainErrorCode {
+    InvalidAddress,
+    InvalidAmount,
+    InvalidPaymentHash,
+    InvalidSwap,
+    SwapAddressNotFound,
+    SwapInProgress,
+    SwapInsufficientFunds,
+    SwapMissingInput,
+    SwapProviderError,
+    UnexpectedAddress,
+    UnexpectedScript,
+}
+
+impl fmt::Display for OnchainErrorCode {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+/// The value error codes
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Serialize)]
+pub enum ValueErrorCode {
+    ConversionFailed,
+    DecodingFailed,
+    InvalidMnemonic,
+    InvalidSignature,
+    InvalidTimestamp,
+    InvalidUrl,
+    MissingParameters,
+    UnexpectedLength,
+    UnexpectedType,
+    UnknownValue,
+}
+
+impl fmt::Display for ValueErrorCode {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
 /// Type of error returned by the SDK
 #[derive(Debug, Error)]
 pub enum SdkError {
```

Comment:
NIT: Enum values in alphabetic order

Reply:
Done [4edbbbe](https://github.com/breez/breez-sdk/pull/469/commits/4edbbbee20fdca35aebe326f3048148e0f9c7b48)